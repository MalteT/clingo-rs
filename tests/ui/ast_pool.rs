use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let args = vec![term1,term2];
    let pool = ast::Pool::new(&args);

    drop(args);
    println!("{:?}",pool);
}