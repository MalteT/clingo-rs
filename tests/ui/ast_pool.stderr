error[E0505]: cannot move out of `args` because it is borrowed
  --> $DIR/ast_pool.rs:9:10
   |
8  |     let pool = ast::Pool::new(&mut args);
   |                               --------- borrow of `args` occurs here
9  |     drop(args);
   |          ^^^^ move out of `args` occurs here
10 |     println!("{:?}",pool);
   |                     ---- borrow later used here

For more information about this error, try `rustc --explain E0505`.
